auth/transports/websocket_transport.go

```
@@ -0,0 +1,222 @@
+	t.mu.Lock()
+	defer t.mu.Unlock()
+
+	if !t.connected || t.conn == nil {
+		return nil // Already disconnected
+	}
+
+	// Signal the receiver goroutine to stop
+	close(t.stopCh)
+
+	// Close the connection
+	err := t.conn.WriteMessage(websocket.CloseMessage, websocket.FormatCloseMessage(websocket.CloseNormalClosure, ""))
+	if err != nil {
+		// Just log the error but continue with disconnect
+		fmt.Printf("Error sending close message: %v\n", err)
+	}
+
+	err = t.conn.Close()
+	t.conn = nil
+	t.connected = false
+
+	return err
+}
+
+// Send sends an AuthMessage via WebSocket
+func (t *WebSocketTransport) Send(message *auth.AuthMessage) error {
+	t.mu.Lock()
+	defer t.mu.Unlock()
+
+	if !t.connected || t.conn == nil {
+		return ErrNotConnected
+	}
+
+	// Marshal the message to JSON
+	jsonData, err := json.Marshal(message)
+	if err != nil {
+		return fmt.Errorf("failed to marshal auth message: %w", err)
+	}
+
+	// Send the message
+	err = t.conn.WriteMessage(websocket.TextMessage, jsonData)
+	if err != nil {
+		t.connected = false
+		t.conn = nil
+		return fmt.Errorf("failed to send WebSocket message: %w", err)
+	}
+
+	return nil
+}
+
+// OnData registers a callback for incoming messages
+func (t *WebSocketTransport) OnData(callback func(*auth.AuthMessage) error) error {
+	t.mu.Lock()
+	defer t.mu.Unlock()
+	t.onDataFuncs = append(t.onDataFuncs, callback)
+	return nil
+}
+
+// IsConnected returns the connection status
+func (t *WebSocketTransport) IsConnected() bool {
+	t.mu.Lock()
+	defer t.mu.Unlock()
+	return t.connected && t.conn != nil
+}
+
+// receiveMessages handles incoming WebSocket messages
+func (t *WebSocketTransport) receiveMessages() {
+	defer func() {
+		t.mu.Lock()
+		t.connected = false
+		t.conn = nil
+		t.mu.Unlock()
+	}()
+
+	for {
+		select {
+		case <-t.stopCh:
+			return // Stopped by disconnect
+		default:
+			// Continue processing
+		}
+
+		// Set read deadline if needed
+		_ = t.conn.SetReadDeadline(time.Now().Add(10 * time.Second))
```

Is ignoring this error safe?